README.md

```diff
@@ -1,70 +1,114 @@
+- `verificationMethods` (map with string keys and values): a set service / public key mappings. the values are public keys `did:key` encoding; they get re-encoded in "multibase" form when rendered in DID document. the key strings should not include a `#` prefix; that will be added when rendering the DID document. used to generate `verificationMethods` of DID document. these keys do not have control over the DID document
+- `alsoKnownAs` (array of strings): priority-ordered list of URIs which indicate other names or aliases associated with the DID identifier
+- `services` (map with string keys; values are maps with `type` and `endpoint` string fields): a set of service / URL mappings. the key strings should not include a `#` prefix; that will be added when rendering the DID document.
 
-type Service = {
-  type: string
-  endpoint: string
-}
-```
+Every update "operation" to the DID identifier, including the initial creation operation ("genesis" operation), contains all of the above information, except for the `did` field. The DID itself is generated from a hash of the signed genesis operation (details described below), which makes the DID entirely self-certifying. Updates after initial creation contain a pointer to the most-recent previous operation (by hash).
 
-The keys specified in the `verificationMethods` object are the "signing keys" used in atproto. The "rotation keys" are used only for control of the DID identity itself. It is permitted to include a key as both a rotation key and a signing key.
+"Operations" are signed and submitted to the central PLC directory server over an un-authenticated HTTP request. The PLC server validates operations against any and all existing operations on the DID (including signature validation, recovery time windows, etc), and either rejects the operation or accepts and permanently stores the operation, along with a server-generated timestamp.
 
-An "operation" object has the following format:
+A special operation type is a "tombstone", which clears all of the data fields and permanently "de-activates" the DID. Note that the usual recovery time window applies to "tombstone" operations.
 
-```ts
-type Operation = {
-  type: 'plc_operation',
-  rotationKeys: string[]
-  verificationMethods: Record<string, string>
-  alsoKnownAs: string[]
-  services: Record<string, Service>
-  prev: CID | null // null if genesis operation
-  sig: string
-}
-```
+Note that `rotationKeys` and `verificationMethods` ("signing keys") may have public keys which are re-used across many accounts. There is not necessarily a one-to-one mapping between a DID and either "rotation" keys or "signing" keys.
+
+Only `secp256k1` ("k256") and NIST P-256 ("p256") keys are currently supported, for both "rotation" and "signing" keys.
+
+### Use with AT Protocol
```

Comment:
I have some questions about if we should leave all the atproto specific stuff in this repo or move over to our repo

but in the mean time, we should document it next tot he code, which is here 👍

Reply:
yup! I did start splitting things out more here, but didn't review the whole doc to be super consistent or separated (had already spent a couple hours on this today).

I think it is fine for them to be entwined for now. if/when folks start actually re-using this for non-atproto stuff we probably need to do a bit of a review of the implementation/spec to ensure that works well, and we can split info out more.